src/NuGet.Core/NuGet.Protocol/Providers/SymbolPackageUpdateResourceV3Provider.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Protocol
+{
+    public class SymbolPackageUpdateResourceV3Provider : ResourceProvider
+    {
+        public SymbolPackageUpdateResourceV3Provider()
+            : base(typeof(SymbolPackageUpdateResourceV3),
+                  nameof(SymbolPackageUpdateResourceV3),
+                  NuGetResourceProviderPositions.Last)
+        {
+        }
+
+        public override async Task<Tuple<bool, INuGetResource>> TryCreate(SourceRepository source, CancellationToken token)
+        {
+            SymbolPackageUpdateResourceV3 symbolPackageUpdateResource = null;
+
+            var serviceIndex = await source.GetResourceAsync<ServiceIndexResourceV3>(token);
+
+            if (serviceIndex != null)
+            {
+                var baseUrl = serviceIndex.GetServiceEntryUri(ServiceTypes.SymbolPackagePublish);
+
+                HttpSource httpSource = null;
+                var sourceUri = baseUrl?.AbsoluteUri;
+                if (!string.IsNullOrEmpty(sourceUri))
+                {
+                    if (!(new Uri(sourceUri)).IsFile)
```

Comment:
this is just inspired from https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Providers/PackageUpdateResourceV3Provider.cs#L38

Reply:
I'd remove it both places because of the following :) 

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Providers/ServiceIndexResourceV3Provider.cs#L49

Reply:
synced up offline. 

This is checking whether the symbols package publish is valid.